lib/FunctionIsolation/IsolateFunctions.cpp

```diff
@@ -549,13 +510,13 @@ void IsolateFunctionsImpl::run() {
   //
   // TODO: we can (and should) push processing of dynamic functions into the
   //       loop emitting individual local functions, and make it lazy
-  T.advance("Create dynamic functions", true);
-  Task DynamicFunctionsTask(Binary.ImportedDynamicFunctions().size(),
-                            "Dynamic functions creation");
+  T->advance("Create dynamic functions", true);
+  DynamicFunctionsTask.emplace(Binary.ImportedDynamicFunctions().size(),
```

Comment:
I don't see why this should become a `class` member.

Reply:
`Task T` is a parent of `DynamicFunctionsTask` and the `Task` object created by the pipewrapper
If `DynamicFunctionsTask` becomes a local variable to the functions it induces a dangling pointer in `Task T` pointing to the now-deleted `DynamicFunctionsTask`